lib/string_map.h

```diff
@@ -0,0 +1,226 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef LIB_STRING_MAP_H_
+#define LIB_STRING_MAP_H_
+
+#include <initializer_list>
+#include <list>
+#include <stdexcept>
+#include <utility>
+
+#include "absl/container/flat_hash_map.h"
+#include "cstring.h"
+
+// Map is ordered by order of element insertion.
```

Comment:
Can you add a more detail comment on when to use this map or what is purpose is? Similar to the description in the PR.

Reply:
Done!